api/src/main/java/org/openmrs/Role.java

```diff
@@ -25,20 +37,49 @@
  *
  * @see Privilege
  */
-public class Role extends BaseChangeableOpenmrsMetadata {
+
+@Entity
+@Table(name="role")
+@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
+public class Role extends BaseOpenmrsObject {
 	
 	public static final long serialVersionUID = 1234233L;
 	
 	private static final Logger log = LoggerFactory.getLogger(Role.class);
 	
-	// Fields
-	
+	@Id
+	@Column(name = "role")
 	private String role;
 	
+	@Transient
+	@Field
+	private String name;
+	
+	@Column(name = "description", length = 255)
+	private String description;
+	
+	@ManyToMany(fetch = FetchType.EAGER)
```

Comment:
I think this doesn't correctly port the cascade settings.

Reply:
Similarly, there seem to be some cache settings on the M-t-M relations I don't see replicated...

Reply:
> I think this doesn't correctly port the cascade settings.

what would be the ideal in this case? i have looked around for a solution that ensures that there's no cascading here

Reply:
I think that's what we actually want (i.e., deleting a role should not delete child roles, inherited roles, or privileges), so maybe just omitting it is fine?

Reply:
Awesome, lemi omit